### PR TITLE
Transaction reseted and NFT deleted only when transaction confirmed

### PR DESCRIPTION
**Pull request by tommasini**
**Description**

Now the collectible it's removed only when the transaction it's confirmed! 

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**https://github.com/MetaMask/metamask-mobile/issues/2784#issuecomment-1038345711**

Solves #2784

### CODE REVIEW

**Comment by tommasini** [comment]
The NFT is deleted after the transaction is confirmed.

It's needed to change the UI.
I propose that the NFT that was transacted stays disabled in the Wallet view, for the user can't send it again while the transaction is happening.

**Comment by gantunesr** [comment]
@tommasini Another scenario to consider is if an user tries to send again the same NFT to a different address with a higher gas fee to overwrite the first transaction. Do you this case is worth taking in consideration?

**Review by gantunesr — CHANGES_REQUESTED** [review]
LGTM but the linter is failing

```
/home/runner/work/metamask-mobile/metamask-mobile/app/components/UI/Transactions/index.test.tsx
Warning:   4:26  warning  Parse errors in imported module '.': ')' expected. ([17](https://github.com/MetaMask/metamask-mobile/runs/5208332772?check_suite_focus=true#step:9:17)0:24)  import/no-named-as-default-member

/home/runner/work/metamask-mobile/metamask-mobile/app/components/Views/SendFlow/Confirm/index.js
Error:   743:10  error    Replace `⏎↹↹↹chainId,⏎↹↹` with `·chainId·`  prettier/prettier
Error:   850:60  error    Delete `⏎↹↹↹↹`                              prettier/prettier
Warning:   851:1   warning  Trailing spaces not allowed                 no-trailing-spaces
```

**Comment by tommasini** [comment]
> Another scenario to consider is if an user tries to send again the same NFT to a different address with a higher gas fee to overwrite the first transaction. Do you think this case is worth taking in consideration?

I think if the transaction is being overwritten that's not a problem because the user will have the information of a bigger gas fee in the confirmation screen.

**Comment by gantunesr** [comment]
@tommasini I was referring to this comment

> It's needed to change the UI.
> I propose that the NFT that was transacted stays disabled in the Wallet view, for the user can't send it again while the transaction is happening.

Because the user might want to send the NFT again using the same nonce and a higher gas fee

**Comment by tommasini** [comment]
> @tommasini I was referring to this comment
> 
> > It's needed to change the UI.
> > I propose that the NFT that was transacted stays disabled in the Wallet view, for the user can't send it again while the transaction is happening.
> 
> Because the user might want to send the NFT again using the same nonce and a higher gas fee

Yes, you are right, We can't disable it because the user might want to overwrite the transaction. When I purpose that I was thinking that the user could not overwrite it.
The only thing that I noticed and I can purpose for the UI It's that could exist in the NFT one label saying the NFT is being transacted (Since one transaction can take some time, like more than 10 minutes). What do you think?

**Comment by gantunesr** [comment]
That's an interesting proposition, we can get some feedback from our designers

**Comment by tommasini** [comment]
> That's an interesting proposition, we can get some feedback from our designers

Thank you! I'm glad to hear that!

**Comment by tommasini** [comment]
This PR will only works when we have the controllers updated with this [PR](https://github.com/MetaMask/controllers/pull/762)

I will leave here a small demo of the expected behavior of these changes in the UI, please let me know what could be improved here

https://user-images.githubusercontent.com/46944231/161263555-cf9d4865-5759-4165-afe9-c42354137c22.mp4

File: app/components/UI/CollectibleModal/index.js [hunk 1]
```diff
@@ -101,8 +112,10 @@ const CollectibleModal = (props) => {
 						collectible={{ ...collectible, contractName }}
 						tradable={isTradable()}
 						onSend={onSend}
+						onViewTransactionDetails={onViewTransactionDetails}
 						openLink={openLink}
 						onTranslation={onCollectibleOverviewTranslation}
+						isTransacting={collectibleUpdated ? !!collectibleUpdated.transactionId : false}
```
Comment (tommasini) [inline app/components/UI/CollectibleModal/index.js:118]: !!collectibleUpdated?.transactionId

**Comment by chrisleewilcox** [comment]
Not able to git pull/checkout branch.....

image

**Comment by tommasini** [comment]
Closing this PR and creating a new one from the metamask remote repo, check [here](https://github.com/MetaMask/metamask-mobile/pull/4091)

**Pull request closed without merging**